login/login.c

```diff
@@ -251,8 +251,9 @@ void login_syslog(glome_login_config_t* config, pam_handle_t* pamh,
 // If the return value is >=0, the buf is NULL-terminated.
 static int read_stdin(char* buf, int buflen) {
```

Comment:
@l9i do you know why we have an `fgets(3p)` implementation in GLOME?

Reply:
It's almost like `fgets` but not quite. The main difference is here: https://github.com/google/glome/blob/5121a2e4657119847729144a62385064c0f79773/login/login.c#L266-L269

`read_stdin` tries to ignore all spurious characters that may appear on a real serial console. One could argue that perhaps a similar effect could be achieved with `fgets` and sanitizing the outbut buffer afterwards (and repeating `fgets` if necessary). I'd be totally fine with changing the implementation or even making this behaviour optional.

Reply:
I see. What kind of characters would we expect to spuriously appear on the console? I believe we need to rely on a `\n` indicating the end of the tag to fix #116.

Reply:
Based on the discussion on the meeting, updated to use `fgets` and return an error if non-printable characters are found. Also always read until the newline to avoid passing input to shell.